doc/contributing/coding-conventions.chapter.md

```diff
@@ -220,7 +220,7 @@ There are a few naming guidelines:
 
 - The `version` attribute _must_ start with a digit e.g`"0.3.1rc2".
 
-- If a package is not a release but a commit from a repository, then the `version` attribute _must_ be the date of that (fetched) commit. The date _must_ be in `"unstable-YYYY-MM-DD"` format.
+- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest _preceding_ version number assigned by upstream, followed by `-unstable-` and the date of the (fetched) commit. The date _must_ be in `"YYYY-MM-DD"` format. E.g. if upstream project `foo` is working on version `2.3` after releasing `2.2`, a commit authored on March 15, 2022 would get `version` of `2.2-unstable-2022-03-15`.
```

Comment:
```suggestion
- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the _latest_ release version number assigned by upstream, followed by `-unstable-` and the date of the commit.
  The date _must_ be in `YYYY-MM-DD` format.
  
  Example: Given a project had its latest release `2.2` in November 2021, a commit authored on March 15, 2022 would have `version = "2.2-unstable-2022-03-15"`.
```

The term "preceding" confuses me, it reads like the one before the latest.

I think this can be worded more concisely without losing information.

Reply:
The reword doesn't make it clear that if Project Whatever released Whatever 2.7 yesterday, but what we package is a commit from the Git history between Whatever 2.5 and 2.6, that the version of our package should be `"2.5-unstable-YYYY-MM-DD"` (the latest _preceding_ version) and not `"2.7-unstable-YYYY-MM-DD"` (the _latest_ version).

Reply:
Ah, I see. The original wording was missing a signal that "preceding" means "preceding the commit in question".

```suggestion
- If a package is a commit from a repository without a version assigned, then the `version` attribute _should_ be the latest release version number preceding that commit, followed by `-unstable-` and the date of the commit.
  The date _must_ be in `YYYY-MM-DD` format.
  
  Example: Given a project had its latest release `2.2` in November 2021, a commit authored on March 15, 2022 would have `version = "2.2-unstable-2022-03-15"`.
```

Reply:
Updated (we do speak of releases and pre-releases and release candidates elsewhere, so I just wrote upstream version instead)